Dockerfile

```diff
@@ -1,4 +1,15 @@
-ARG IMAGE=python:3.9-alpine
+ARG IMAGE=alpine:3.17
+FROM ${IMAGE} as builder
+
+WORKDIR /build
+
+# Install dependencies
+RUN apk add --no-cache git python3 python3-dev py3-pip py3-wheel build-base gcc libffi-dev openssl-dev musl-dev cargo
```

Comment:
I suggest using the `RUN --mount type=cache` command to optimize the apk cache, as documented here:
https://docs.docker.com/build/cache/#keep-layers-small